src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -219,6 +219,24 @@ private static ConstantValue AsConstOrDefault(BoundExpression init)
             return ConstantValue.Default(type.SpecialType);
         }
 
+        /// <summary>
+        /// Determine if enum arrays can be initialized using block initialization.
+        /// </summary>
+        /// <returns>True if it's safe to use block initialization for enum arrays.</returns>
+        /// <remarks>
+        /// In NetFx 4.0, block array initializers do not work on all combinations of {32/64 X Debug/Retail} when array elements are enums.
+        /// This is fixed in 4.5 thus enabling block array initialization for a very common case.
+        /// We look for the presence of <see cref="System.Runtime.GCLatencyMode.SustainedLowLatency"/> which was introduced in .NET Framework 4.5
+        /// </remarks>
+        private bool EnableEnumArrayBlockInitialization
```

Comment:
I'm not seeing what change requires this to move from the existing location?

Reply:
> I'm not seeing what change requires this to move from the existing location?

There is only one consumer of this API, the emit layer. This makes it clear what it should do about tracking well-known types. The method is moved to avoid accidental use elsewhere in the future. 

---
In reply to: [351367773](https://github.com/dotnet/roslyn/pull/40010#discussion_r351367773) [](ancestors = 351367773)